be/src/exprs/vectorized/time_functions.cpp

```diff
@@ -1454,13 +1457,16 @@ ColumnPtr TimeFunctions::datetime_format(FunctionContext* context, const Columns
     if (fc != nullptr && fc->is_valid) {
         return do_format<TYPE_DATETIME>(fc, columns);
     } else {
+        bool all_const = ColumnHelper::is_all_const(columns);
         ColumnBuilder<TYPE_VARCHAR> builder;
         ColumnViewer<TYPE_DATETIME> viewer_date(columns[0]);
         ColumnViewer<TYPE_VARCHAR> viewer_format(columns[1]);
 
+        size_t num_rows = all_const ? viewer_date.size() : columns[0]->size();
```

Comment:
Could you please add some comment for this line?
Because I don't know why.

Reply:
all all input was const column, we could use `viewer_date.size()` instead of columns[0]->size().
then viewer_date.size() will return 1 or 0, we could reduce unnecessary calculations

Reply:
In order to let other know, please add this reason the the code as comment.

Reply:
done